### PR TITLE
Support private repositories

### PR DESCRIPTION
**Pull request by xuchunyang**
Github Application token is needed to enable this feature,

also see docstring of `helm-github-stars-token' for more info.

### CODE REVIEW

**Comment by xuchunyang** [comment]
I had enabled and disabled this feature, In either case, it worked fine.

I did not write test for two new functions because I thought that is unnecessary.

For now, 3 variables (`hgs/helm-c-source-stars`, `hgs/helm-c-source-repos` and `hgs/helm-c-source-private-repos`) are almost identical, we need to find a way to merge them, do you have any ideas about this?

**Comment by Sliim** [comment]
Wow really cool!!! :)
Sorry for my latency, I'm currently on multiple projects :s.

I tested your code, seems to works fine, but I can't validate that private repositories are correctly displayed because I havn't private repos on github, so I will trust you :).

I think we can use the token for all requests on Github when user set `helm-github-stars-token` no?

**Comment by Sliim** [comment]
I fixed travis on `master` branch, you can rebase your code to fix this travis build status ;)

**Comment by xuchunyang** [comment]
> I think we can use the token for all requests on Github when user set helm-github-stars-token no?

Of course, we can, I'll do it now.

**Comment by xuchunyang** [comment]
I merged private repos to repos because it was a bit easier to implement.

**Comment by xuchunyang** [comment]
I just rebase your master branch and force-push. I hope I didn't do anything wrong.

File: helm-github-stars.el [hunk 1]
```diff
@@ -178,26 +208,46 @@ When disabled (nil) don't align description."
...
-    (let ((next-request t)
-          (current-page 1))
-      (while next-request
-        (let ((response (hgs/parse-github-response (hgs/request-github-repos current-page))))
-          (if (= 0 (length response))
-              (setq next-request nil)
-            (progn
-              (setq repos-list (vconcat repos-list response))
-              (setq current-page (1+ current-page)))))))
+    (if helm-github-stars-token
```
Comment (Sliim) [inline helm-github-stars.el]: There is a mistake here! When `helm-github-stars-token` is provided, only first github page is fetched, so repository list is incomplete. I'm looking how to improve this..

File: helm-github-stars.el [hunk 2]
```diff
@@ -178,26 +208,46 @@ When disabled (nil) don't align description."
...
+        (progn
+          ;; Fetch user's starred repositories
```
Comment (Sliim) [inline helm-github-stars.el]: What's the profit to use the token for starred repositories? Result is same as `https://api.github.com/users/Sliim/starred?` no?

**Comment by Sliim** [comment]
Pending PR: https://github.com/xuchunyang/helm-github-stars/pull/2

Enabling / disabling the feature have same effects for me, my starred & public repos are correctly collected, but I can't test for private repos, I need your help :).

Feel free to improve my code.

File: helm-github-stars.el (hunk 1 above)
Comment (Sliim) [inline helm-github-stars.el]: Should be ok with https://github.com/xuchunyang/helm-github-stars/pull/2

File: helm-github-stars.el (hunk 2 above)
Comment (xuchunyang) [inline helm-github-stars.el]: The result is same, it just takes advantage of the access token.

**Comment by xuchunyang** [comment]
Now it should be able to fetch all starred repos and all user's repos. Only one of username and token is necessary, if both are provided, token will be used.

The main purpose of using token is to fetch user's private repos.

**Comment by xuchunyang** [comment]
@Sliim Can all your starred repos be displayed by using the token?

I think I had solved that problem in my last commit, but I didn't test because both the number of my starred repos and my own repos is <100. (Actually, it was Ok when I set `per_page=8`)

**Comment by Sliim** [comment]
Your code looks good, but I'm not able to test it now. I will test your change as soon as possible and send you a feedback ;)

**Comment by Sliim** [comment]
Works fine :+1: merged!
Good job & Thanks :)

**Pull request merged**